server/sonar-server/src/test/java/org/sonar/server/qualitygate/ws/GetByProjectActionTest.java

```diff
@@ -129,80 +136,143 @@ public void default_quality_gate() {
+    assertThat(result.getQualityGate().getName()).isEqualTo(qualityGate.getName());
+  }
+
+  @Test
+  public void default_organization_is_used_when_no_organization_parameter() {
+    OrganizationDto organization = db.getDefaultOrganization();
+    QGateWithOrgDto qualityGate = db.qualityGates().insertQualityGate(organization);
+    ComponentDto project = db.components().insertPrivateProject(organization);
+    userSession.logIn().addProjectPermission(UserRole.USER, project);
+    associateProjectToQualityGate(project.getId(), qualityGate.getId());
+
+    GetByProjectResponse result = ws.newRequest()
+      .setParam("project", project.getKey())
+      .executeProtobuf(GetByProjectResponse.class);
+
+    assertThat(result.getQualityGate().getName()).isEqualTo(qualityGate.getName());
+  }
+
+  @Test
+  public void fail_when_project_does_not_not_belong_to_organization() {
+    OrganizationDto organization = db.organizations().insert();
+    OrganizationDto otherOrganization = db.organizations().insert();
+    ComponentDto project = db.components().insertPrivateProject(otherOrganization);
+    userSession.logIn().addProjectPermission(UserRole.USER, project);
+    QGateWithOrgDto qualityGate = db.qualityGates().insertQualityGate(organization);
+    associateProjectToQualityGate(project.getId(), qualityGate.getId());
+
+    expectedException.expect(NotFoundException.class);
+    expectedException.expectMessage(String.format("Project '%s' doesn't exist in organization '%s'", project.getKey(), organization.getKey()));
+
+    ws.newRequest()
+      .setParam("project", project.getKey())
+      .setParam("organization", organization.getKey())
+      .execute();
   }
 
   @Test
   public void fail_when_insufficient_permission() {
-    ComponentDto project = componentDb.insertComponent(ComponentTesting.newPrivateProjectDto(db.getDefaultOrganization()));
+    OrganizationDto organization = db.organizations().insert();
+    ComponentDto project = db.components().insertPrivateProject(organization);
+    QGateWithOrgDto qualityGate = db.qualityGates().insertQualityGate(organization);
+    associateProjectToQualityGate(project.getId(), qualityGate.getId());
     userSession.logIn();
-    QualityGateDto dbQualityGate = insertQualityGate("Sonar way");
-    setDefaultQualityGate(dbQualityGate.getId());
 
     expectedException.expect(ForbiddenException.class);
 
-    callByKey(project.getKey());
+    ws.newRequest()
+      .setParam("project", project.getKey())
+      .setParam("organization", organization.getKey())
+      .execute();
   }
 
   @Test
   public void fail_when_project_does_not_exist() {
+    OrganizationDto organization = db.organizations().insert();
+
     expectedException.expect(NotFoundException.class);
 
-    callByKey("Unknown");
+    ws.newRequest()
+      .setParam("project", "Unknown")
+      .setParam("organization", organization.getKey())
+      .execute();
   }
 
```

Comment:
I think a test that check that built-in is not destroyed is missing.

Reply:
no pb, please add it